src/parser.cpp

```diff
@@ -377,11 +377,10 @@ namespace ada::parser {
           ada_log("HOST ", helpers::substring(url_data, input_position));
 
           std::string_view host_view = helpers::substring(url_data, input_position);
-          bool inside_brackets{false};
-          size_t location = helpers::get_host_delimiter_location(url, host_view, inside_brackets);
+          auto [location, found_colon] = helpers::get_host_delimiter_location(url.is_special(), host_view);
```

Comment:
Even though I think std::tuple is the correct choice here, we tend to use to mutable parameters in other places. Can we add a TODO to refactor those parts to follow the same pattern here?